libs/parallelism/algorithms/include/hpx/parallel/algorithms/partition.hpp

```diff
@@ -118,14 +569,42 @@ namespace hpx { namespace parallel { inline namespace v1 {
 
                         std::rotate(first, mid, last);
 
-                        // for some library implementations std::rotate
-                        // does not return the new middle point
-                        std::advance(first, std::distance(mid, last));
-                        return first;
-                    },
-                    std::move(left), std::move(right));
+                        // for some library implementations std::rotate
+                        // does not return the new middle point
+                        std::advance(first, std::distance(mid, last));
+                        return first;
+                    },
+                    std::move(left), std::move(right));
+            }
+        };
+
+        template <typename BidirIter, typename Sent, typename F, typename Proj>
+        static BidirIter stable_partition_seq(
```

Comment:
@hkaiser This algorithm has a O(n) space complexity since we need to store the false values. The standard library implementations also have a less efficient O(nlogn) algorithm that they use if sufficient space is not available.

Should we add the other algorithm too? If so, how do we check if sufficient space is available?

Reply:
We can do that, I'd suggest leaving it to a separate PR, however. 

Re how to check: try and catch `std::bad_alloc` exceptions ;-) How is that handled in the std libraries?

Reply:
`We can do that, I'd suggest leaving it to a separate PR, however.`
Oh ok sure.
`Re how to check: try and catch std::bad_alloc exceptions ;-)`
Thanks!
`How is that handled in the std libraries?`
They try and allocate a temporary buffer, if the length of the returned buffer is not enough then they revert to the inplace implementation.